packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,228 @@
+import { FC, useEffect, useState } from 'react';
+import { Skeleton, Table, formatTokenAmount } from '@sifi/shared-ui';
+import { Token } from '@sifi/sdk';
+import { Button } from 'src/components/Button';
+import { PartnerTokensByChain, TokenOfPartner } from 'src/hooks/usePartnerTokens';
+import { getChainById, getChainIcon } from 'src/utils/chains';
+import { useAccount, useNetwork, useSwitchNetwork } from 'wagmi';
+import { useFetchTokens } from 'src/hooks/useFetchTokens';
+import { usePartnerWithdraw } from 'src/hooks/usePartnerWithdraw';
+import { ConnectWallet } from 'src/components/ConnectWallet/ConnectWallet';
+import { firstAndLast, getEvmTxUrl } from 'src/utils';
+import { ETH_CONTRACT_ADDRESS, ETH_ZERO_ADDRESS } from 'src/constants';
+import { usePartnerData } from 'src/hooks/usePartnerData';
+
+const extractTokenAddress = (id: string): string => {
+  // The id is a concatenation of two Ethereum addresses,
+  // the token address is last 40 characters of the id.
+  return '0x' + id.slice(-40);
+};
+
+type PartnerTokensTableProps = {
+  partnerTokensByChain: PartnerTokensByChain;
+};
+
+type TokenRowProps = {
+  token: TokenOfPartner;
+  chainId: string;
+  handleWithdraw: (id: string, chainId: number) => void;
+  withdrawnTokens: { [key: string]: string };
+  tokensByChainIdAndAddress: Record<string, Token>;
+};
+
+const TokenRow: FC<TokenRowProps> = ({
+  token,
+  chainId,
+  handleWithdraw,
+  withdrawnTokens,
+  tokensByChainIdAndAddress,
+}) => {
+  let tokenAddress = extractTokenAddress(token.id);
+  if (tokenAddress === ETH_ZERO_ADDRESS) {
```

Comment:
I see you figured this out. The backend API uses `0xEEE` but the smart contracts use `0x000` for the native token. The backend translates the `0xEEE` into `0x000` for the contract calls. The zero address cannot be checksummed, while the `0xEEE` one can.